suave/cstore/engine2.go

```diff
@@ -0,0 +1,283 @@
+package cstore
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	suave "github.com/ethereum/go-ethereum/suave/core"
+	"github.com/google/uuid"
+
+	"github.com/gorilla/mux"
+)
+
+type Engine2 struct {
```

Comment:
Can we give this a more descriptive name?  Maybe `NonAuthenticatedEngine`?  I think a docstring might be helpful, too.

Ditto for `NewEngine2`.